main.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2020 Sebastian Pipping <sebastian@pipping.org>
+ * Licensed under AGPL v3 or later
+ */
+package main
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/hartwork/go-wait-for-it/cli"
+	"github.com/hartwork/go-wait-for-it/logging"
+	"github.com/hartwork/go-wait-for-it/syntax"
+)
+
+type ConnectResult struct {
+	address  syntax.Address
+	duration time.Duration
+	success  bool
+}
+
+func waitForAddress(address syntax.Address) chan bool {
+	available := make(chan bool)
+	go func() {
+		for {
+			c, err := net.Dial("tcp", address.String())
+			if err == nil {
+				available <- true
+				c.Close()
+				break
+			}
+			time.Sleep(500 * time.Millisecond)
+		}
+	}()
+	return available
+}
+
+func waitForAddressWithTimeout(address syntax.Address, timeout time.Duration, results chan ConnectResult) {
```

Comment:
The timeout can actually be more nicely implemented with a context.Context that carries a timeout via `context.WithTimeout(context.Background(), time.Second)`.
This is totally fine to do in a separate PR as improvement :blush:

Reply:
Interesting! I'll go read https://blog.golang.org/context

Reply:
I grabbed some ideas from the post and other docs on contexts by now but I'm still too much in the dark with some key questions about it, to use it myself in code today. I think I'll delay the topic of contexts in Go until a moment with voice chat on the topic, in the future. Thanks for bringing it up, definitely of interest for any future HTTP server of mine in Go :smiley: